astropy/table/serialize.py

```diff
@@ -144,18 +145,21 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
         if not has_info_class(data, MixinInfo):
             col_cls = MaskedColumn if (hasattr(data, 'mask')
                                        and np.any(data.mask)) else Column
-            new_cols.append(col_cls(data, name=new_name, **new_info))
-            obj_attrs[data_attr] = SerializedColumn({'name': new_name})
-        else:
-            # recurse. This will define obj_attrs[new_name].
-            _represent_mixin_as_column(data, new_name, new_cols, obj_attrs)
-            obj_attrs[data_attr] = SerializedColumn(obj_attrs.pop(new_name))
+            data = col_cls(data, name=new_name, **new_info)
```

Comment:
Change is needed because we can no longer be sure that a `Column` will not need to be serialized. I think this was actually a buglet before, since for `MaskedColumn`, one can never be sure it will not need to be serialized (since the user may want 'data_mask'). But it was not easy to expose that bug before.